sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2403,6 +2408,17 @@ pi_result cuda_piKernelCreate(pi_program program, const char *kernel_name,
     retErr = PI_OUT_OF_HOST_MEMORY;
   }
 
+  pi_device device = program->get_context()->get_device();
```

Comment:
Note.
While it's OK for now, this code assumes that there is only one device per context which might not be true eventually.

Reply:
BTW, now, the device is not currently involved in looking for the value of `PI_KERNEL_GROUP_INFO_COMPILE_WORK_GROUP_SIZE`, but I left your note as a comment.